### ISSUE TITLE
Changing passwords reset API access_tokens

### ISSUE THREAD

**Issue by abraham** [appdotnet/api-spec#82]
One of the features of using OAuth is you can change passwords and existing apps continue to work. It would be nice if access_tokens didn't reset unless users revoke the.

**Comment by jschlesser** [comment]
Id like to hear more discussion on this.  I agree that if you have a lot of apps, reauth can be a major hassle.  It might be the intent though.  Maybe an option on password reset to force auth tokens to reset?  I can see both sides, its a a matter of intention.  Whats reasonable?

**Comment by adamjgrant** [comment]
I don't know if Twitter is a dirty word here, but what about doing it the same way as they do? I'm pretty sure nothing is reset, keys or secrets.

**Comment by abitgone** [comment]
In my opinion, passwords and auth tokens should be completely separate beasts – one should have no effect on the other.

That said, I think @jschlesser's idea is good: when you change your password, an option would be presented to also revoke all your existing auth tokens, in case your account had become compromised and other auth tokens had been generated during that time. 

This would be a welcome addition to some, just as an option to revoke all your existing auth tokens without having to change your password.

**Comment by swhitley** [comment]
I agree with the need for separation and the desire to prompt token refreshes.  From what I've seen so far, tokens don't expire; and I think that's a good thing.  The Google route, with expired tokens, is overly complicated.
